lib/socket_proxy/receiver_supervisor.ex

```diff
@@ -7,16 +7,12 @@ defmodule SocketProxy.ReceiverSupervisor do
   end
 
   def start_child(arg) do
-    spec = %{id: SocketProxy.Receiver, start: {SocketProxy.Receiver, :start_link, [arg]}}
+    spec = %{id: SocketProxy.Receiver, start: {SocketProxy.Receiver, :start_link, [arg]}, restart: :temporary}
```

Comment:
Ah! Makes sense why it started happening with the latest `socket_proxy` upgrade. I changed the `spec` around a bit to use `DynamicSupervisor` since the old `:simple_one_for_one` strategy was deprecated in the newer version of Elixir that we switched to.